src/Symfony/Component/Routing/Exception/MissingMandatoryParametersException.php

```diff
@@ -19,4 +19,38 @@
  */
 class MissingMandatoryParametersException extends \InvalidArgumentException implements ExceptionInterface
 {
+    private array $missingParameters = [];
+    private string $routeName = '';
+
+    /**
+     * @param string $routeName
+     * @param array  $missingParameters
+     * @param int    $code
+     */
+    public function __construct(string $routeName = '', $missingParameters = null, /*int*/ $code = 0, \Throwable $previous = null)
+    {
+        if (\is_array($missingParameters)) {
+            $this->missingParameters = $missingParameters;
+            $this->routeName = $routeName;
+            $message = sprintf('Some mandatory parameters are missing ("%s") to generate a URL for route "%s".', implode('", "', $missingParameters), $routeName
+            );
+        } else {
+            trigger_deprecation('symfony/routing', '6.1', 'Construction of "%s" with an exception message is deprecated, provide the route name and an array of missing parameters instead.', __CLASS__);
+            $message = $routeName;
+            $code = (int) $missingParameters;
+            $previous = $code instanceof \Throwable ? $code : null;
```

Comment:
this must be *before* overriding the `$code` variable

Reply:
this could be covered by a second legacy test using the full legacy signature to pass a previous exception

Reply:
done & done ! thanks !